3rd_training.lua

```diff
@@ -2320,8 +2325,16 @@ function on_gui()
     end
 
     -- input history
-    if training_settings.display_p1_input_history then input_history_draw(input_history[1], 4, 49, false) end
-    if training_settings.display_p2_input_history then input_history_draw(input_history[2], screen_width - 4, 49, true) end
+    if training_settings.display_p1_input_history_dynamic and training_settings.display_p1_input_history then
+      if player.pos_x < 320 then
```

Comment:
`player` is not always P1, for instance when you are in recording in mode, player and dummy are swapped between P1 and P2, so it's probably better to use `player_objects[1]` than `player` here.
It's a minor thing so I'll merge the PR and do it myself to avoid useless back and forth, but I just wanted you to know.